receiver/splunkhecreceiver/receiver.go

```diff
@@ -208,38 +241,66 @@ func (r *splunkReceiver) handleReq(resp http.ResponseWriter, req *http.Request)
 			return
 		}
 		if msg.IsMetric() {
-			// Currently unsupported.
-			r.failRequest(ctx, resp, http.StatusBadRequest, errUnsupportedMetricEvent, err)
+			if r.metricsConsumer == nil {
+				r.failRequest(ctx, resp, http.StatusBadRequest, errUnsupportedMetricEvent, err)
+				return
+			}
+		} else if r.logsConsumer == nil {
+			r.failRequest(ctx, resp, http.StatusBadRequest, errUnsupportedLogEvent, err)
 			return
 		}
 
-		messagesReceived++
 		events = append(events, &msg)
 	}
+	if r.logsConsumer != nil {
+		r.consumeLogs(ctx, events, resp, req)
+	} else {
+		r.consumeMetrics(ctx, events, resp, req)
+	}
+}
 
-	customizer := func(resource pdata.Resource) {}
+func (r *splunkReceiver) createResourceCustomizer(req *http.Request) func(pdata.Resource) {
 	if r.config.AccessTokenPassthrough {
-		if accessToken := req.Header.Get(splunk.SplunkHECTokenHeader); accessToken != "" {
-			customizer = func(resource pdata.Resource) {
-				resource.Attributes().InsertString(splunk.SplunkHecTokenLabel, accessToken)
+		if accessToken := req.Header.Get(splunk.HECTokenHeader); accessToken != "" {
+			return func(resource pdata.Resource) {
+				if resource.IsNil() {
+					resource.InitEmpty()
+				}
+				resource.Attributes().InsertString(splunk.HecTokenLabel, accessToken)
 			}
 		}
 	}
+	return func(resource pdata.Resource) {}
+}
+
+func (r *splunkReceiver) consumeMetrics(ctx context.Context, events []*splunk.Event, resp http.ResponseWriter, req *http.Request) {
+	md, _ := SplunkHecToMetricsData(r.logger, events, r.createResourceCustomizer(req))
```

Comment:
Is error handling intentionally silently omitted here?

Reply:
Now, it's numOfDroppedMetrics, a value only used in testing.

Reply:
What if `SplunkHecToMetricsData` returns an error for some other reason? It will be ignored here.

Reply:
`SplunkHecToMetricsData` does not return an error object at this time. It's just an int counter.

Reply:
Sorry, my mistake.